### PR TITLE
Replace retry with roko

### PR DESCRIPTION
**Pull request by moskyb**
We recently extracted the retry module from the agent into its own go module, [roko](https://github.com/buildkite/roko).

This PR removes the existing retry library, and replaces it with roko. There's no functionality change between the two modules other than the module name.

### CODE REVIEW

File: clicommand/pipeline_upload.go [hunk 1]
```diff
@@ -320,7 +320,7 @@ var PipelineUploadCommand = cli.Command{
 
 			return err
 			// On a server error, it means there is downtime or other problems, we
-			// need to retry. Let's retry every 5 seconds, for a total of 5 minutes.
+			// need to roko. Let's retry every 5 seconds, for a total of 5 minutes.
```
Comment (tessereth) [inline clicommand/pipeline_upload.go:323]: Ah.... Maybe not this one, as good as it sounds 😅

**Review by tessereth — APPROVED** [review]

File: clicommand/pipeline_upload.go (hunk 1 above)
Comment (moskyb) [reply clicommand/pipeline_upload.go:323]: 🙈 omg

**Comment by pda** [comment]
image

🙌🏼

**Comment by sj26** [comment]
Nice 🙌

**Pull request merged**